### PR TITLE
Support System Upgrade.

### PR DESCRIPTION
**Pull request by HitoriWoB**
Upgraded support system as discussed.

Support type statics not being used everywhere
Added functionality for staff to re-open supports
Corrected bug where comments wouldn't go through when closing or opening a support

### CODE REVIEW

File: classes/SupportManager.php [hunk 1]
```diff
@@ -96,115 +91,91 @@ public function __construct($system, $user_id = 0, $user_support_level = false,
...
+                self::$TYPE_BALANCE,
+                self::$TYPE_REPORT_STAFF,
+                self::$TYPE_APPEAL_BAN,
+                self::$TYPE_SUGGESTION,
+                self::$TYPE_MISC,
             ];
-            if($staffLevel == System::SC_MODERATOR) {
-                $types[] = 'Mod Request';
+
+            if($this->user->staff_level == System::SC_MODERATOR) {
```
Comment (levimeahan) [inline classes/SupportManager.php:115]:
```suggestion
            if($this->user->isModerator()) {
```

File: classes/SupportManager.php [hunk 2]
```diff
@@ -385,163 +262,158 @@ public function supportSearch($variables = [], $limit = false, $order = [], $off
...
-     */
-    public function updateSupport($support_id, $admin_id = 0, $admin_name = '', $staff_level=0) {
-        // Validate support exists and belongs to user
-        if($this->admin) {
-            $support_data = $this->fetchSupportByID($support_id);
-        }
-        else {
-            $support_data = $this->fetchSupportByID($support_id, $this->user_id);
-        }
+    public function updateSupport($support_id, $support_staff = false) {
```
Comment (levimeahan) [inline classes/SupportManager.php:398]: Could we replace the `$support_staff` parameter with `$this->user->isSupportStaff()`?

Comment (HitoriWoB) [reply classes/SupportManager.php:398]: Will change this to `$this->staff` as that dictates if the instance of SupportManager is a staff connection or a user connection. 

My thought process here is:
Oreo has a support request entered and is support staff. Replies on the user end, I'd think sending a PM that her support was updated could be confusing.

**Comment by HitoriWoB** [comment]
Made changes as reviewed.

**Comment by HitoriWoB** [comment]
Upgrade is now ready to go.

File: classes/SupportManager.php (hunk 2 above)
Comment (levimeahan) [reply classes/SupportManager.php]: That makes sense - We can go ahead and merge this, but we should update that variable to reflect that. Possible names:

`$this->in_staff_mode`
`$this->viewing_as_staff`

**Review by levimeahan — APPROVED** [review]

**Pull request merged**